

# Exercise One - Reasearch Methods

## 3. Use and Explore the Data in Stata

In [1]:
cd 

C:\Users\Rafael\Eco 727 - Research Methods\Analysis


In [2]:
 use "C:\Users\Rafael\Eco 727 - Research Methods\Data\cps_00002.dta"

### Tabulate mish, eligorg, and labforce. 

In [3]:
tab1 mish eligorg labforce 


-> tabulation of mish  

   month in |
    sample, |
  household |
      level |      Freq.     Percent        Cum.
------------+-----------------------------------
        one |     82,939        1.23        1.23
        two |     83,727        1.24        2.46
      three |     83,525        1.23        3.70
       four |  3,126,792       46.21       49.90
       five |     83,941        1.24       51.14
        six |     84,009        1.24       52.39
      seven |     83,888        1.24       53.62
      eight |  3,138,231       46.38      100.00
------------+-----------------------------------
      Total |  6,767,052      100.00

-> tabulation of eligorg  

  (earnings) |
 eligibility |
        flag |      Freq.     Percent        Cum.
-------------+-----------------------------------
    eligible |  6,767,052      100.00      100.00
-------------+-----------------------------------
       Total |  6,767,052      100.00

-> tabulation of labforce  

        labor force status | 

### Which rotation groups don’t belong in these data?

If we look at the tabulation of mish, we can observe that all rotation groups except for 4 and 8 don't belong in these data. 

### The wrong rotation groups are from what year or years?

To find out the year from where the wrong rotation groups are from, we can tabulate mish by year to observe in what year these observations are recorded.  

In [4]:
tab mish year if inrange(year, 1982, 1986)


  month in |
   sample, |
 household |                      survey year
     level |      1982       1983       1984       1985       1986 |     Total
-----------+-------------------------------------------------------+----------
       one |    82,939          0          0          0          0 |    82,939 
       two |    83,727          0          0          0          0 |    83,727 
     three |    83,525          0          0          0          0 |    83,525 
      four |    83,073     82,847     84,024     85,677     85,364 |   420,985 
      five |    83,941          0          0          0          0 |    83,941 
       six |    84,009          0          0          0          0 |    84,009 
     seven |    83,888          0          0          0          0 |    83,888 
     eight |    83,577     82,328     84,427     85,937     84,902 |   421,171 
-----------+-------------------------------------------------------+----------
     Total |   668,679    165,175    168,451    17

We can see that the wrong rotation groups are from year 1982. As such we only keep 4 and 8.  

### If mish isn’t 4 or 8, drop these observations

In [5]:
keep if mish==4|mish==8

(502,029 observations deleted)


In [6]:
tabulate mish


   month in |
    sample, |
  household |
      level |      Freq.     Percent        Cum.
------------+-----------------------------------
       four |  3,126,792       49.91       49.91
      eight |  3,138,231       50.09      100.00
------------+-----------------------------------
      Total |  6,265,023      100.00


Once we tabulate mish again, we see that we only keep the rotation groups we wanted.

### (a) Describing the data set

In [7]:
describe


Contains data from C:\Users\Rafael\Eco 727 - Research Methods\Data\cps_00002.dta
  obs:     6,265,023                          
 vars:            29                          05 JAN 2021 23:48
--------------------------------------------------------------------------------
              storage   display    value
variable name   type    format     label      variable label
--------------------------------------------------------------------------------
year            int     %8.0g                 survey year
serial          long    %12.0g                household serial number
month           byte    %8.0g      MONTH      month
hwtfinl         double  %12.0g                household weight, basic monthly
cpsid           double  %12.0g                cpsid, household record
asecflag        byte    %8.0g      ASECFLAG   flag for asec
mish            byte    %8.0g      MISH       month in sample, household level
region          byte    %8.0g      REGION     region and division
pernum    

We can see that we have roughly 6.26 million observations on 29 variables. 

### (b) Listing 5 observations of 8-10 variables at a time

In [8]:
list year-mish in 1/5, noobs
list region-empstat in 1/5, noobs
list labforce-earnwt in 1/5, noobs
list qearnwee-eligorg in 1/5, noobs



  +--------------------------------------------------------------------+
  | year   serial     month   hwtfinl         cpsid   asecflag    mish |
  |--------------------------------------------------------------------|
  | 1982        5   january         .   1.98110e+13          .    four |
  | 1982        5   january         .   1.98110e+13          .    four |
  | 1982        7   january         .   1.98110e+13          .    four |
  | 1982       10   january         .   1.98010e+13          .   eight |
  | 1982       10   january         .   1.98010e+13          .   eight |
  +--------------------------------------------------------------------+


  +----------------------------------------------------------------------------+
  |   region   pernum    wtfinl        cpsidp   age      sex    race   empstat |
  |----------------------------------------------------------------------------|
  | west nor        1   1009.79   1.98110e+13    52     male   white   at work |
  | west nor   

Based on the listings, we can observe that a lot of the variables have missing observations. Variables such as: hwtfinl, asecflag, qearnwee, etc. 

### (c) Summarizing the data

In [9]:
summarize


    Variable |        Obs        Mean    Std. Dev.       Min        Max
-------------+---------------------------------------------------------
        year |  6,265,023    2000.534    11.13821       1982       2020
      serial |  6,265,023     34528.6    20251.51          1      74625
       month |  6,265,023    6.506208     3.45006          1         12
     hwtfinl |  5,087,446    2129.075    1180.058          0   34716.21
       cpsid |  6,265,023    2.00e+13    1.11e+11   1.98e+13   2.02e+13
-------------+---------------------------------------------------------
    asecflag |    518,026           2           0          2          2
        mish |  6,265,023    6.003652    1.999997          4          8
      region |  6,265,023    27.22327    10.76149         11         42
      pernum |  6,265,023    1.676862     .937971          1         22
      wtfinl |  6,265,023    2075.533    1173.679      22.24   34716.21
-------------+-------------------------------------------------

Based on the summary we can see that that some variables have missing observations as well as see that some variable don't have any variability from looking at their standard deviations.

### (d) Tabulating the categorical variables

In [10]:
tab1 region sex race empstat classwkr educ qearnwee paidhour union 



-> tabulation of region  

        region and division |      Freq.     Percent        Cum.
----------------------------+-----------------------------------
       new england division |    591,859        9.45        9.45
   middle atlantic division |    752,144       12.01       21.45
east north central division |    861,009       13.74       35.20
west north central division |    655,772       10.47       45.66
    south atlantic division |  1,074,806       17.16       62.82
east south central division |    315,910        5.04       67.86
west south central division |    537,229        8.58       76.44
          mountain division |    650,937       10.39       86.83
           pacific division |    825,357       13.17      100.00
----------------------------+-----------------------------------
                      Total |  6,265,023      100.00

-> tabulation of sex  

        sex |      Freq.     Percent        Cum.
------------+-----------------------------------
       male |  3

------------+-----------------------------------
      Total |  6,265,023      100.00

-> tabulation of union  

                 union membership |      Freq.     Percent        Cum.
----------------------------------+-----------------------------------
                              niu |         16        0.00        0.00
                no union coverage |  5,000,173       84.27       84.27
            member of labor union |    832,906       14.04       98.31
covered by union but not a member |    100,103        1.69      100.00
----------------------------------+-----------------------------------
                            Total |  5,933,198      100.00


### Confirming the structure of the data

In [11]:
tabulate month mish


           |   month in sample,
           |    household level
     month |      four      eight |     Total
-----------+----------------------+----------
   january |   260,270    261,606 |   521,876 
  february |   261,872    261,726 |   523,598 
     march |   258,948    259,078 |   518,026 
     april |   259,600    258,855 |   518,455 
       may |   258,330    257,839 |   516,169 
      june |   261,958    261,951 |   523,909 
      july |   261,346    263,884 |   525,230 
    august |   260,687    262,905 |   523,592 
 september |   260,444    262,688 |   523,132 
   october |   263,670    265,440 |   529,110 
  november |   263,095    264,376 |   527,471 
  december |   256,572    257,883 |   514,455 
-----------+----------------------+----------
     Total | 3,126,792  3,138,231 | 6,265,023 


We confirm the structure of the data by computing cross-tabulations of month and rotation group (mish).

### Dropping variables 

In [12]:
drop asecflag 
drop labforce 
drop eligorg

We drop these variables as they don't vary.

### Finding year American Indians and Asians were first included in race

To find in what year are American Indians and Asians first included in race, we can tabulate race by year to observe in what year these observations were first recorded.  

In [13]:
tab race year if inrange(year, 1982, 1992)


                      |                 survey year
                 race |      1982       1983       1984       1985 |     Total
----------------------+--------------------------------------------+----------
                white |   146,627    145,099    147,612    150,014 | 1,621,413 
          black/negro |    14,669     14,722     15,322     16,027 |   173,954 
american indian/aleut |         0          0          0          0 |     5,383 
asian or pacific isla |         0          0          0          0 |    20,267 
other (single) race,  |     5,354      5,354      5,517      5,573 |    40,683 
----------------------+--------------------------------------------+----------
                Total |   166,650    165,175    168,451    171,614 | 1,861,700 


                      |                 survey year
                 race |      1986       1987       1988       1989 |     Total
----------------------+--------------------------------------------+----------
                wh

We can observe that American Indians and Asians were first included in race in 1989.

## 4. Clean the data

### Addressing the coding of missing observations

In [14]:
replace region=. if region==97
replace sex=. if sex==9
replace race=. if race==999
replace empstat=. if empstat==00
replace occ1950 =. if inlist(occ1950, 997, 999) // Unknown, Unemployed- last worked over x years ago
replace ind1950=. if inlist(ind1950, 000, 997,998) // NIU, Nonclassifiable, Industry not reported
replace classwkr=. if inlist(classwkr, 00, 99) // NIU, Missing/Unknown
replace uhrswork1=. if uhrswork1==999 & inrange(year, 1989, 1993) // Missing/Unknown
replace uhrswork1=. if inlist(uhrswork1, 997, 999) & year>=1994 // Hours vary, Missing/Unknown 
replace educ=. if inlist(educ, 000, 001, 999) // NIU or no schooling, NIU or blank, Missing/Unknown
replace hourwage=. if hourwage == 999.99
replace paidhour=. if paidhour==0
replace union=. if union==0
replace earnweek =. if earnweek == 9999.99
replace uhrsworkorg=. if inlist(uhrsworkorg, 998, 999) // Don't Know, NIU 


(0 real changes made)

(0 real changes made)

(0 real changes made)

(0 real changes made)

(0 real changes made)

(0 real changes made)

(0 real changes made)

(0 real changes made)

(253,746 real changes made, 253,746 to missing)

(0 real changes made)

(2,567,522 real changes made, 2,567,522 to missing)

(0 real changes made)

(16 real changes made, 16 to missing)

(8,239 real changes made, 8,239 to missing)

(2,362,969 real changes made, 2,362,969 to missing)


### Recoding for consistency accross years

In [15]:
global eyear=2020

In [16]:
numlabel EDUC, add // to add code number for labels to help me with recoding educ.
tabulate educ




          educational attainment recode |      Freq.     Percent        Cum.
----------------------------------------+-----------------------------------
                   2. none or preschool |     11,309        0.18        0.18
               10. grades 1, 2, 3, or 4 |     19,651        0.31        0.49
                            11. grade 1 |        827        0.01        0.51
                            12. grade 2 |      2,023        0.03        0.54
                            13. grade 3 |      3,228        0.05        0.59
                            14. grade 4 |      4,010        0.06        0.66
                      20. grades 5 or 6 |     46,348        0.74        1.39
                            21. grade 5 |      5,261        0.08        1.48
                            22. grade 6 |     13,252        0.21        1.69
                      30. grades 7 or 8 |     47,427        0.76        2.45
                            31. grade 7 |     11,324        0.18        2

### Recoding Education

We first recode the education variable educ to create the variable grade, which measures highest grade completed. We use Jaeger (1997) to transform degrees into grades.

In [17]:
generate grade=.
*** 1982-1991
  replace grade=0  if educ==2 & year<=1991 // (none or preschool)
  replace grade=1  if educ==11 & year<=1991
  replace grade=2  if educ==12 & year<=1991
  replace grade=3  if educ==13 & year<=1991
  replace grade=4  if educ==14 & year<=1991
  replace grade=5  if educ==21 & year<=1991
  replace grade=6  if educ==22 & year<=1991
  replace grade=7  if educ==31 & year<=1991
  replace grade=8  if educ==32 & year<=1991
  replace grade=9  if educ==40 & year<=1991
  replace grade=10 if educ==50 & year<=1991
  replace grade=11 if educ==60 & year<=1991
  replace grade=12 if (educ==72 | educ==73) & year<=1991 // 72 is `diploma unclear'
  replace grade=13 if educ==80 & year<=1991
  replace grade=14 if educ==90 & year<=1991
  replace grade=15 if educ==100 & year<=1991
  replace grade=16 if educ==110 & year<=1991
  replace grade=17 if educ==121 & year<=1991
  replace grade=18 if educ==122 & year<=1991
  
  *** 1992-2109 (Jaeger 1997)
  replace grade=0   if educ==2  & year>=1992
  replace grade=2.5 if educ==10 & year>=1992 // grades 1-4
  replace grade=5.5 if educ==20 & year>=1992 // grades 5-6
  replace grade=7.5 if educ==30 & year>=1992 // grades 7-8
  replace grade=9   if educ==40 & year>=1992 // grade 9
  replace grade=10  if educ==50 & year>=1992 // grade 10
  replace grade=11  if educ==60 & year>=1992 // grade 11
  replace grade=12  if (educ==71 | educ==73) & year>=1992 // 71 is `no diploma'
  replace grade=13  if educ==81 & year>=1992 // some college, no degree
  replace grade=14  if (educ==91 | educ==92) & year>=1992 // academic or vocational associate's degree
  replace grade=16  if educ==111 & year>=1992
  replace grade=18  if inlist(educ, 123, 124, 125) & year>=1992
  order grade, before(educ)


(6,265,023 missing values generated)

(3,693 real changes made)

(827 real changes made)

(2,023 real changes made)

(3,228 real changes made)

(4,010 real changes made)

(5,261 real changes made)

(13,252 real changes made)

(11,324 real changes made)

(37,861 real changes made)

(38,411 real changes made)

(56,240 real changes made)

(60,845 real changes made)

(704,348 real changes made)

(134,820 real changes made)

(168,063 real changes made)

(63,319 real changes made)

(229,636 real changes made)

(45,334 real changes made)

(110,245 real changes made)

(7,616 real changes made)

(19,651 real changes made)

(46,348 real changes made)

(47,427 real changes made)

(52,713 real changes made)

(69,429 real changes made)

(104,965 real changes made)

(1,429,154 real changes made)

(904,609 real changes made)

(458,194 real changes made)

(949,397 real changes made)

(482,780 real changes made)



### Recoding Occupation

We procceed by recoding occupation. We generate occuply to replace occ1950 amd we code for 10 occupations.

In [18]:
generate occuply=.
  
  *** 1982-2020
  replace occuply=1 if inrange(occ1950, 000, 099) & inrange(year, 1982, $eyear) // professional, technical
  replace occuply=2 if inrange(occ1950, 100, 290) & inrange(year, 1982, $eyear) // managers, officials, propreitors
  replace occuply=3 if inrange(occ1950, 300, 390) & inrange(year, 1982, $eyear) // clerical, kindred
  replace occuply=4 if inrange(occ1950, 400, 490) & inrange(year, 1982, $eyear) // sales
  replace occuply=5 if inrange(occ1950, 500, 594) & inrange(year, 1982, $eyear) // craftsmen
  replace occuply=6 if inrange(occ1950, 600, 690) & inrange(year, 1982, $eyear) // operatives
  replace occuply=7 if inrange(occ1950, 700, 720) & inrange(year, 1982, $eyear) // service (private hh)
  replace occuply=8 if inrange(occ1950, 730, 790) & inrange(year, 1982, $eyear) // service (other)
  replace occuply=9 if inrange(occ1950, 810, 970) & inrange(year, 1982, $eyear) // laborers
  replace occuply=10 if occ1950==036              & inrange(year, 1982, $eyear) // editors and reporters
  tab year occ1950 if occ1950~=. & occuply==. & inrange(year, 1982, $eyear)


(6,265,023 missing values generated)

(1,491,531 real changes made)

(780,556 real changes made)

(1,147,993 real changes made)

(364,059 real changes made)

(617,963 real changes made)

(690,773 real changes made)

(51,113 real changes made)

(827,347 real changes made)

(293,688 real changes made)

(21,928 real changes made)

no observations


### Recoding Industry

We procceed by recoding industry. We generate industly to replace ind1950 amd we code for 12 industries.

In [19]:
generate industly=.

*** 1982-2019
  replace industly=1 if inrange(ind1950, 105, 239) & inrange(year, 1982, $eyear) // agriculture, fisheries, mining
  replace industly=2 if inrange(ind1950, 246, 246) & inrange(year, 1982, $eyear) // construction
  replace industly=3 if inrange(ind1950, 306, 399) & inrange(year, 1982, $eyear) // durable manufacturing
  replace industly=4 if inrange(ind1950, 406, 499) & inrange(year, 1982, $eyear) // nondurable manufacturing
  replace industly=5 if inrange(ind1950, 506, 598) & inrange(year, 1982, $eyear) // communications, transportation, utilities
  replace industly=6 if inrange(ind1950, 606, 627) & inrange(year, 1982, $eyear) // wholesale trade
  replace industly=7 if inrange(ind1950, 636, 699) & inrange(year, 1982, $eyear) // retail trade
  replace industly=8 if inrange(ind1950, 716, 746) & inrange(year, 1982, $eyear) // professional services, fire
  replace industly=8 if inrange(ind1950, 868, 879) & inrange(year, 1982, $eyear) // professional services, fire
  replace industly=8 if inrange(ind1950, 897, 899) & inrange(year, 1982, $eyear) // professional services, fire
  replace industly=9 if inlist(ind1950, 888, 896)  & inrange(year, 1982, $eyear) // education, welfare services
  replace industly=10 if inrange(ind1950, 906, 936) & inrange(year, 1982, $eyear) // public administration
  replace industly=11 if inrange(ind1950, 806, 859) & inrange(year, 1982, $eyear) // Advertising, Miscellaneous entertainment and recreation services
  replace industly=12 if ind1950==679              & inrange(year, 1982, $eyear) // Eating and drinking places
  tab year ind1950 if ind1950~=. & industly==. & inrange(year, 1982, $eyear)


(6,265,023 missing values generated)

(152,383 real changes made)

(357,177 real changes made)

(594,329 real changes made)

(382,442 real changes made)

(365,320 real changes made)

(203,873 real changes made)

(1,020,359 real changes made)

(412,657 real changes made)

(702,448 real changes made)

(187,940 real changes made)

(779,792 real changes made)

(400,396 real changes made)

(705,907 real changes made)

(313,932 real changes made)

no observations


We drop redudant variables

In [20]:
drop occ1950
drop ind1950

We define labels for the occupation and industry variables 

In [21]:
  label define OCCUP 1 "professional, technical"  2  "managers, officials, propreitors"  3  "clerical, kindred" ///
                     4 "sales"  5  "craftsmen"    6  "operatives"  7  "service (private hh)" ///
                     8  "service (other)"         9  "laborers"   10 "editors and reporters"
  label value occup OCCUP

  label define INDUST 1 "agriculture, fisheries, mining"  2  "construction"  3  "durable manufacturing" ///
                      4 "nondurable manufacturing"  5 "communications, transportation, utilities"  6  "wholesale trade" ///
                      7 "retail trade"  8  "professional services, fire"  9 "education, welfare services" ///
                     10 "public administration"  11 "Advertising, Miscellaneous entertainment and recreation services" 12 "Eating and drinking places"
  label value indust INDUST

We quickly tabulate occup and indust to see our results. 

In [22]:
tab1 occup indust


-> tabulation of occuply  

                         occuply |      Freq.     Percent        Cum.
---------------------------------+-----------------------------------
         professional, technical |  1,469,603       23.46       23.46
managers, officials, propreitors |    780,556       12.46       35.92
               clerical, kindred |  1,147,993       18.32       54.24
                           sales |    364,059        5.81       60.05
                       craftsmen |    617,963        9.86       69.91
                      operatives |    690,773       11.03       80.94
            service (private hh) |     51,113        0.82       81.76
                 service (other) |    827,347       13.21       94.96
                        laborers |    293,688        4.69       99.65
           editors and reporters |     21,928        0.35      100.00
---------------------------------+-----------------------------------
                           Total |  6,265,023      100.00

->

###  Create a variable to identify top-coded values of earnweek

In [23]:
gen topcode_earnweek = .
replace topcode_earnweek = 0 if earnweek != . & inrange(year, 1982,$eyear)
replace topcode_earnweek = 1 if earnweek == 999 & inrange(year, 1982, 1988)
replace topcode_earnweek = 1 if earnweek== 1923 & inrange(year, 1989, 1997)
replace topcode_earnweek = 1 if earnweek == 2884.61 & inrange(year, 1998,$eyear)
label variable topcode_earnweek "Top Coded Values of Weekly Earnings" // To rewrite the heading
label define TOPCODE_EARNWEEK 0 "0-Not Top Coded" 1 "1-Top Coded" // To rename labels, so it's not just 1 and 0 
label value topcode_earnweek TOPCODE_EARNWEEK
tab year topcode_earnweek, missing


(6,265,023 missing values generated)

(6,256,784 real changes made)

(34,917 real changes made)

(14,628 real changes made)

(78,835 real changes made)





           |    Top Coded Values of Weekly
    survey |             Earnings
      year | 0-Not Top  1-Top Cod          . |     Total
-----------+---------------------------------+----------
      1982 |   164,140      2,481         29 |   166,650 
      1983 |   161,905      3,236         34 |   165,175 
      1984 |   164,421      3,998         32 |   168,451 
      1985 |   166,619      4,958         37 |   171,614 
      1986 |   164,392      5,844         30 |   170,266 
      1987 |   164,413      6,809         25 |   171,247 
      1988 |   156,560      7,591         23 |   164,174 
      1989 |   166,790        822          0 |   167,612 
      1990 |   175,175      1,080          0 |   176,255 
      1991 |   170,053      1,243          0 |   171,296 
      1992 |   167,648      1,312          0 |   168,960 
      1993 | 

We replace the topcode variable to zero only if the earnweek varaible is not missing. As such the first line of the code incorporates this.

## 5. Explore the Edited Data

### Describe, summarize, and tabulate variables in the edited data

In [24]:
describe


Contains data from C:\Users\Rafael\Eco 727 - Research Methods\Data\cps_00002.dta
  obs:     6,265,023                          
 vars:            28                          05 JAN 2021 23:48
--------------------------------------------------------------------------------
              storage   display    value
variable name   type    format     label      variable label
--------------------------------------------------------------------------------
year            int     %8.0g                 survey year
serial          long    %12.0g                household serial number
month           byte    %8.0g      MONTH      month
hwtfinl         double  %12.0g                household weight, basic monthly
cpsid           double  %12.0g                cpsid, household record
mish            byte    %8.0g      MISH       month in sample, household level
region          byte    %8.0g      REGION     region and division
pernum          byte    %8.0g                 person number in sample 

This time around we have roughly 2.65 million observations with 28 variables. We can see that we added the variable topcode_earnweek.

In [25]:
summarize


    Variable |        Obs        Mean    Std. Dev.       Min        Max
-------------+---------------------------------------------------------
        year |  6,265,023    2000.534    11.13821       1982       2020
      serial |  6,265,023     34528.6    20251.51          1      74625
       month |  6,265,023    6.506208     3.45006          1         12
     hwtfinl |  5,087,446    2129.075    1180.058          0   34716.21
       cpsid |  6,265,023    2.00e+13    1.11e+11   1.98e+13   2.02e+13
-------------+---------------------------------------------------------
        mish |  6,265,023    6.003652    1.999997          4          8
      region |  6,265,023    27.22327    10.76149         11         42
      pernum |  6,265,023    1.676862     .937971          1         22
      wtfinl |  6,265,023    2075.533    1173.679      22.24   34716.21
      cpsidp |  6,265,023    2.00e+13    1.11e+11   1.98e+13   2.02e+13
-------------+-------------------------------------------------

Once we summarize the data we can observe the varaible that we edited and how they changed. For example for the variable grade we can see that it ranges from 0 to 18. 

In [26]:
tab1 mish grade occuply industly 


-> tabulation of mish  

   month in |
    sample, |
  household |
      level |      Freq.     Percent        Cum.
------------+-----------------------------------
       four |  3,126,792       49.91       49.91
      eight |  3,138,231       50.09      100.00
------------+-----------------------------------
      Total |  6,265,023      100.00

-> tabulation of grade  

      grade |      Freq.     Percent        Cum.
------------+-----------------------------------
          0 |     11,309        0.18        0.18
          1 |        827        0.01        0.19
          2 |      2,023        0.03        0.23
        2.5 |     19,651        0.31        0.54
          3 |      3,228        0.05        0.59
          4 |      4,010        0.06        0.66
          5 |      5,261        0.08        0.74
        5.5 |     46,348        0.74        1.48
          6 |     13,252        0.21        1.69
          7 |     11,324        0.18        1.87
        7.5 |     47,427        0.7

Here we tabulate some of the varaibles we created in order to check for consistency.

### What percentage of weekly wage observations are top coded?

In [27]:
tab topcode_earnweek, missing


      Top Coded |
      Values of |
Weekly Earnings |      Freq.     Percent        Cum.
----------------+-----------------------------------
0-Not Top Coded |  6,128,404       97.82       97.82
    1-Top Coded |    128,380        2.05       99.87
              . |      8,239        0.13      100.00
----------------+-----------------------------------
          Total |  6,265,023      100.00


We see that in the total data set, 2.05 percent of weekly wages are top coded.

### In what year is top-coding of earnweek most common?

In [28]:
tab year topcode_earnweek, missing


           |    Top Coded Values of Weekly
    survey |             Earnings
      year | 0-Not Top  1-Top Cod          . |     Total
-----------+---------------------------------+----------
      1982 |   164,140      2,481         29 |   166,650 
      1983 |   161,905      3,236         34 |   165,175 
      1984 |   164,421      3,998         32 |   168,451 
      1985 |   166,619      4,958         37 |   171,614 
      1986 |   164,392      5,844         30 |   170,266 
      1987 |   164,413      6,809         25 |   171,247 
      1988 |   156,560      7,591         23 |   164,174 
      1989 |   166,790        822          0 |   167,612 
      1990 |   175,175      1,080          0 |   176,255 
      1991 |   170,053      1,243          0 |   171,296 
      1992 |   167,648      1,312          0 |   168,960 
      1993 |   165,307      1,407          0 |   166,714 
      1994 |   160,220      2,039        486 |   162,745 
      1995 |   159,620      2,222        419 |   162,2

Based on frequency, we notice that it was in year 1988 were top coding was more common

In [30]:
save "../Data/cps", replace

file ../Data/cps.dta saved


In [31]:
tab year


survey year |      Freq.     Percent        Cum.
------------+-----------------------------------
       1982 |    166,650        2.66        2.66
       1983 |    165,175        2.64        5.30
       1984 |    168,451        2.69        7.99
       1985 |    171,614        2.74       10.72
       1986 |    170,266        2.72       13.44
       1987 |    171,247        2.73       16.18
       1988 |    164,174        2.62       18.80
       1989 |    167,612        2.68       21.47
       1990 |    176,255        2.81       24.28
       1991 |    171,296        2.73       27.02
       1992 |    168,960        2.70       29.72
       1993 |    166,714        2.66       32.38
       1994 |    162,745        2.60       34.97
       1995 |    162,261        2.59       37.56
       1996 |    144,803        2.31       39.88
       1997 |    147,540        2.35       42.23
       1998 |    149,278        2.38       44.61
       1999 |    151,471        2.42       47.03
       2000 |    15